pkg/libmount/parse.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2020 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package libmount
+
+import (
+	"errors"
+	"strings"
+)
+
+var (
+	InvalidTokenError error = errors.New("invalid token error")
+)
+
+type Parser interface {
+	Parse(string) (*Filesystem, error)
+}
+
+type mountInfoParser struct{}
+type mountsParser struct{}
+
+func NewMountsParser() Parser {
+	return mountsParser{}
+}
+
+func NewMountInfoParser() Parser {
+	return mountInfoParser{}
+}
+
+func NewParser(format MountTabFormat) Parser {
+	switch format {
+	case MNT_FMT_MOUNTINFO:
+		return NewMountInfoParser()
+	default:
+		return NewMountsParser()
+	}
+}
+
+// parse one line of mountinfo file
+func (p mountInfoParser) Parse(line string) (*Filesystem, error) {
+	return nil, errors.New("not implemented")
+}
+
+// parse one line of mounts/{fs, m}tab file
+func (p mountsParser) Parse(line string) (*Filesystem, error) {
```

Comment:
we are not parsing mtab or fstab right?

Reply:
So the same function is used to parse proc/mounts, mtab, and fstab file. The C library does the same.